packages/pcd/gpc-pcd/src/GPCPCDPackage.ts

```diff
@@ -290,19 +303,121 @@ function validateInputPOD(
+    // Return false if some entry in the config is not in the POD
+    if (
+      configEntries.some((entryName) => podEntries[entryName] === undefined)
+    ) {
+      return false;
+    }
+
+    // Alist of (entry name, membership list name) pairs.
+    const listMembershipAlist = Object.entries(podConfig.entries).flatMap(
+      ([entryName, entryConfig]) =>
+        entryConfig.isMemberOf !== undefined
+          ? [[entryName, entryConfig.isMemberOf]]
+          : []
+    );
+
+    // If there are list membership checks in the proof config *and* the
+    // serialised lists are passed in as a parameter, check whether any
+    // do not pass.
+    if (
+      listMembershipAlist.length > 0 &&
+      params.membershipLists !== undefined
+    ) {
+      // Deserialise membership Lists.
+      let membershipLists: PODMembershipLists;
+
+      try {
+        membershipLists = podMembershipListsFromSimplifiedJSON(
+          params.membershipLists
+        );
+      } catch (_) {
+        params.notFoundMessage = "Error deserialising membership lists.";
+        return false;
+      }
+
+      if (
+        listMembershipAlist.some(
+          ([entryName, listName]) =>
+            !membershipLists[listName]
+              .map((value) => applyOrMap(podValueHash, value))
+              .includes(podValueHash(podEntries[entryName]))
+        )
+      ) {
+        return false;
+      }
+    }
+
+    // Check for prescribed values.
+    if (params.prescribedValues !== undefined) {
+      let prescribedValues: GPCPCDPrescribedPODValues;
+      try {
+        prescribedValues = gpcPCDPrescribedPODValuesFromSimplifiedJSON(
+          params.prescribedValues
+        );
+      } catch (e) {
+        if (e instanceof TypeError) {
+          params.notFoundMessage = e.message;
+          return false;
+        }
+        throw e;
+      }
+
+      const prescribedValuesForPod = prescribedValues[podName];
+
+      if (prescribedValuesForPod !== undefined) {
+        // Check signer's public key.
+        if (
+          prescribedValuesForPod.signerPublicKey !== undefined &&
+          !_.isEqual(
+            decodePublicKey(prescribedValuesForPod.signerPublicKey),
+            decodePublicKey(podPCD.pod.signerPublicKey)
+          )
+        ) {
+          return false;
+        }
+
+        // Check POD for prescribed entries (if any).
+        if (prescribedValuesForPod?.entries !== undefined) {
+          const prescribedEntryAlist = Object.entries(
+            prescribedValuesForPod.entries
+          );
+
+          // Sanity check: All prescribed entry names should appear in the config
+          // and their values should be revealed!
+          for (const [entryName, _] of prescribedEntryAlist) {
+            if (podConfig.entries[entryName] === undefined) {
+              params.notFoundMessage =
+                "Invalid prescribed entry record: Not all entries are present in the proof configuration.";
```

Comment:
These kinds of checks should really be able to be checked client-side by the requester.  Maybe there should be a separate exported `checkPrescribedEntries` function?